packages/vuetify/src/effects/color.ts

```diff
@@ -0,0 +1,79 @@
+import { computed, ComputedRef, isRef, Ref } from 'vue'
+
+export interface ColorProps {
+  color?: string
+}
+
+export const colorProps = (defaults: Partial<ColorProps> = {}) => ({
+  color: {
+    type: String,
+    default: defaults.color,
+  },
+})
+
+const isCssColor = (color: string): boolean => !!color && !!color.match(/^(#|(rgb|hsl)a?\()/)
+const isProps = <T>(props: T | RefValue): props is T => !isRef(props)
+
+type PropValue = string | null | undefined
+type RefValue = Readonly<Ref<PropValue>>
+type Color = { class: Record<string, boolean> } | { style: Record<string, string> } | {}
```

Comment:
First i though that `ComputedRef<Color>` is a ref for color ('#000' string)
`ColorData` is imho more accurate name